server/src/main/java/org/apache/iotdb/db/metadata/MGraph.java

```diff
@@ -99,11 +120,11 @@ public MNode addDeviceIdToMTree(String deviceId) throws PathErrorException {
    * Add a seriesPath to {@code PTree}.
    */
   void addPathToPTree(String path) throws PathErrorException {
-    String[] nodes = path.trim().split(DOUB_SEPARATOR);
+    String[] nodes = getNodeNames(path);
     if (nodes.length == 0) {
       throw new PathErrorException("Timeseries is null.");
     }
-    String rootName = path.trim().split(DOUB_SEPARATOR)[0];
+    String rootName = getNodeNames(path)[0];
```

Comment:
Thanks for your suggestions!

The `getNodeNames()` method is extracted to the class `MetaUtils`. I modified the method according to your advice to reduce the times of calling `split()` method.

I modified the code to use saved result instead of calling the method.